### DISCUSSION TITLE
Is nvidia graphical acceleration still an supported feature?

### DISCUSSION THREAD

**Discussion by Sandr-aa** [neutrinolabs/xorgxrdp#317] [Q&A]
The gfx_mainline_merge branch, also packaged in aur as [xorgxrdp-nvidia](https://aur.archlinux.org/packages/xorgxrdp-nvidia) is needed for graphical acceleration according to the archwiki entry for xrdp. Seeing that the gfx_mainline_merge branch hasn't been updated in 10 months and it not working in my case, I have a few questions:
Is the nvidia implementation discontinued/stalled/not feasible anymore?
In case this only broke because of an version incompatibility, what packages could be downgraded to fix this?

**Comment by matt335672** [comment]
@Sandr-aa - GFX itself has now been merged in this project so the gfx_mainline_merge branch won't see any more updates.

NVidia support is being worked on sporadically (see some of the comments in #2383), but it still isn't officially supported by this project yet. I can't answer your downgrade question.

**Comment by jsorg71** [comment]
mainline_merge is being retired as it's too difficult to maintain.
The latest NVidia bits are at
https://github.com/jsorg71/xrdp/tree/nvenc
https://github.com/jsorg71/xorgxrdp/tree/nvenc
These are based off latest devel and much easier to merge.
I'll update the other issues.

**Reply by moetayuko** [reply]
Hello @jsorg71, I've been testing your nvenc branches recently. One drawback I noticed so far is that dynamic resolution is broken: the remote session becomes black after resizing. Toggling the XRDP_USE_ACCEL_ASSIST option (ig it controls whether to use nvenc when streaming) in /etc/xrdp/sesman.ini doesn't make any difference.

My client is Remmina and the feature works fine with the outdated https://github.com/Nexarian/xrdp/tree/mainline_merge https://github.com/Nexarian/xorgxrdp/tree/mainline_merge branches. Could you take a look into the issue?

**Reply by auggie246** [reply]
@jsorg71 base on your latest commits last week, what should be XRDP_NVENC_CFLAGS and XRDP_NVENC_LIBS flags?

edit: answered my own question
`XRDP_NVENC_CFLAGS="-I/usr/local/cuda-12.4/include -I$HOME/Video_Codec_SDK_12.2.72/Interface" XRDP_NVENC_LIBS="-lcuda -lnvidia-encode" ./configure --enable-vsock --enable-jpeg --enable-fuse --enable-pixman --enable-opus --enable-mp3lame --enable-x264 --enable-nvenc`

**Comment by effeffe** [comment]
I'm kinda necrobumping this thread, as I want to share my experience with the main branch of xorgxrdp and nvidia.

I initially was trying to achieve full acceleration directly on nvidia, but I have realised that glamour works just fine on an nvidia 1080 with proprietary drivers. I'm using RHEL 9.5, so the drivers might be not as up-to-date as other systems, but it's okay.

I have been able to run a system with multiple logins from me and other people, all with a pretty reliable remote xorg session using KDE.
NGL, it was a PAIN to set up, as I had multiple occasions where the session would just not start, but I am able to watch videos on youtube with them being fluid. Maybe the only thing that might be improved upon is latency, but it's already very usable.
Only thing I have to compile myself that was not in the repositories is pipewire support (lie, I compiled these branches between 10 and 15 times when I was trying to get it up and running, but I eventually reverted back to the repository installation)
